### PR TITLE
Add crosslink tool

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** Adds the [crosslink](https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/crosslink) tool to simplify the function of adding replace statements to intra-repository go modules. 

_Please delete paragraphs that you did not use before submitting._

### CODE REVIEW

**Comment by bryan-aguilar** [comment]
Can this PR get a skip changelog label?

File: Makefile [hunk 1]
```diff
@@ -433,3 +434,10 @@ checklinks:
 	command -v markdown-link-check >/dev/null 2>&1 || { echo >&2 "markdown-link-check not installed. Run 'npm install -g markdown-link-check'"; exit 1; }
 	find . -name \*.md -print0 | xargs -0 -n1 \
 		markdown-link-check -q -c ./.github/workflows/check_links_config.json || true
+
+# error message "failed to sync logger:  sync /dev/stderr: inappropriate ioctl for device"
+# is a known issue but does not affect function. 
+.PHONY: crosslink
+crosslink: install-tools
```
Comment (bogdandrutu) [inline Makefile:441]: we don't require install-tools in other places (unless I forget), just try to be consistent.

**Review by bogdandrutu — COMMENTED** [review]
Lets make this step part of github action "check" and if git diffs then fail.

File: Makefile (hunk 1 above)
Comment (bryan-aguilar) [reply Makefile:441]: I believe there is some inconsistency. Multimod requires `install-tools`. But I will remove it to match the majority.

**Comment by bryan-aguilar** [comment]
Updated GHA and make directive.

I also added the `--prune` flag as default. Since this will be running on PR build it should continue to clean the modules clean. You can see the details of the prune command [here](https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/crosslink#prune---prune---p).

**Pull request merged**